### DISCUSSION TITLE
can it be used with a SN74HC165 is a 8-Bit Parallel-Load Shift Registers

### DISCUSSION THREAD

**Discussion by sakearzoo** [bxparks/AceButton#110] [Q&A]
As i have limited pins with my esp32 i am have to use Shift Registers to read 16 push button switch (non-latching switch). is it possible to use this library for this ?

**Comment by bxparks** [comment]
The library's framework is definitely capable of supporting this. But you'd have to create a new subclass of `ButtonConfig` (call it something like `ShiftButtonConfig`), and implement the code to read the shift registers and process the buttons that match the corresponding pin on the shift register.

It'd be a great addition to the library. If you get it working, consider sending me a PR.

BTW, it looks like the 74HC165 is not a tristate device, so it must be the only device on the bus. But it looks like the 74HC589 does support tristate:
https://electronics.stackexchange.com/questions/121249/how-to-put-a-74hc165-on-an-spi-bus

**Reply by bxparks** [reply]
Oh I see, the 74HC165 is cheap and easy to find. The 74HC589 is more expensive and more difficult to buy.
